common/build/eslint-config-fluid/src/custom-rules/no-member-release-tags.js

```diff
@@ -0,0 +1,84 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const { TSDocParser } = require("@microsoft/tsdoc");
+
+/**
+ *
+ * @param comment: String - The comment included inside the class / interface / type.
+ * @returns hasReleaseTag: Boolean - True; if the comment contains any release tags.
+ */
+function hasReleaseTag(comment) {
+	const parser = new TSDocParser();
+	const parserContext = parser.parseString(comment);
+
+	const hasReleaseTag =
+		parserContext.docComment.modifierTagSet.isAlpha() ||
+		parserContext.docComment.modifierTagSet.isBeta() ||
+		parserContext.docComment.modifierTagSet.isPublic() ||
+		parserContext.docComment.modifierTagSet.isInternal();
+
+	return hasReleaseTag;
+}
+
+/**
+ * Helper function abstracted for reusability for different types of node.
+ *
+ * @param node - An AST node which is begin traversed.
+ * @param context - The context object containing information that is relevant to the context of the rule.
+ * {@link https://eslint.org/docs/latest/extend/custom-rules}
+ */
+function errorLoggerHelper(node, context) {
+	const sourceCode = context.sourceCode;
+	const comments = sourceCode.getCommentsBefore(node);
+
+	comments.forEach((comment) => {
+		// ESLint trims the asterisk of the comment while TSDocParser expects the original format of the comment block.
+		const formattedComment = `/** ${comment.value} */`;
+		if (hasReleaseTag(formattedComment)) {
+			context.report({
+				node: node,
+				// `node` object has different strucutre based on the AST scope type.
+				message: `Including the release-tag for ${node.key.name} at line ${
+					node.key.loc.start.line
+				} in ${
+					node.parent.parent.id
+						? node.parent.parent.id.name
+						: node.parent.parent.parent.id.name
```

Comment:
Is this guaranteed to work without having to go "further up"? Why can we trust that `node.parent.parent.id` or `node.parent.parent.parent.id` will always be what we expect? Also, why is the first case not using just `node.parent`?

Reply:
As per our conversation yesterday, I am also concerned of the "extent" of the traversal in `node` object's `parent` objects.  

The pattern I found so far is that the `class expression` needs an extra step of `parent` traversal inside the `node` object. 

Since the structure of the AST object can vastly vary depending on the implementation detail, there is a chance that the current logic may not work for several edge cases. I added a comment above the `context.report()` method explaining the reason behind checking `node.parent.parent.id` prior to logging, and if our developers find edge cases in the future, I think we can adjust the logic accordingly. 

However, I think the current test cases (e.g., class expression, class declaration, abstract class, interface, and type) cover most of the development pattern that our linter should be concerned of.